src/combinations.rs

```diff
@@ -31,18 +31,56 @@ impl<I> fmt::Debug for Combinations<I>
 pub fn combinations<I>(iter: I, k: usize) -> Combinations<I>
     where I: Iterator
 {
-    let mut pool: LazyBuffer<I> = LazyBuffer::new(iter);
+    Combinations::new(iter, k)
+}
+
+impl<I: Iterator> Combinations<I> {
+    #[inline]
+    pub(crate) fn new(iter: I, k: usize) -> Self {
```

Comment:
I'm thinking maybe this constructor should be put inline into combinations::combinations() function?